editors/zed/src/slint.rs

```
@@ -49,10 +45,20 @@ impl SlintExtension {
             &zed::LanguageServerInstallationStatus::CheckingForUpdate,
         );
 
-        let release = zed::latest_github_release(
-            "slint-ui/slint",
-            zed::GithubReleaseOptions { require_assets: true, pre_release: false },
-        )?;
+        let release_tag = concat!("v", env!("CARGO_PKG_VERSION"));
+        let release = match zed::github_release_by_tag_name("slint-ui/slint", release_tag) {
+            Ok(release) => release,
+            Err(e) => {
+                // The release don't exist? Maybe hasn't been released yet and we can try the nightly version
```

if it doesn't exists, it may be better to let it fail with a nice message. I would make `nightly` opt-in

---

Yup, failing with a message is a better idea than failing silently.

---

Thanks for the review.
You're right that silently falling back to nightly is not ideal.
The goal is to be able to test the extension before the release, from the repository, as the extension in the repo would likely have the next version already.
I wonder if there is a way to detect if we are in "dev mode"

>  I would make nightly opt-in

Any suggestion on what mechanism to use to make it "opt in"?

---

What about a setting?

This is how I've configured this extension to pick up `slint-lsp` binary from within the nix development shell: 

https://github.com/woile/nix-config/blob/d0684376c33baff959a26ec6e5011caa793687f0/programs/zed-editor/settings.json#L79-L83

And here's where it's read from the LspSettings

https://github.com/slint-ui/slint/blob/9c71f0339a2b1844c3480257df05a87491ffd333/editors/zed/src/slint.rs#L23-L25

Maybe it can be refactor to:

```rs
struct SlintBinary {
    path: String,
    args: Option<Vec<String>>,
}

struct SlintLspSettings {
    binary: SlintBinary,
    nightly: bool
}
```

---

@xunicatt I like this suggestion, but it doesn't seem like we can access env variable unfortunately

---

> and how to set the extension tonightly?

What do you mean? in Zed's `settings.json`

```
"slint-lsp": {
      "binary": {
        "path": "slint-lsp"
      },
     "nightly": true
    },
```

---

@woile  Okay got it.

@ogoffart Yeah we can do that using `zed_extension_api::Worktree::shell_env` which returns `Vec<(String, String)` containing Key-Value pair for env vars. And I did a little test with `FOO=1 zed` and the variable showed up with my other env variables.